pkg/openstackclient/funcs.go

```diff
@@ -34,76 +41,83 @@ func ClientPod(
 		},
 	}
 
-	var terminationGracePeriodSeconds int64 = 0
-	runAsUser := int64(42401)
-	runAsGroup := int64(42401)
+	envVars := map[string]env.Setter{}
+	envVars["OS_CLOUD"] = env.SetValue("default")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+
 	clientPod.ObjectMeta = metav1.ObjectMeta{
 		Name:      instance.Name,
 		Namespace: instance.Namespace,
 		Labels:    labels,
 	}
-	clientPod.Spec.TerminationGracePeriodSeconds = &terminationGracePeriodSeconds
+	clientPod.Spec.TerminationGracePeriodSeconds = ptr.To[int64](0)
 	clientPod.Spec.ServiceAccountName = instance.RbacResourceName()
-	clientPod.Spec.Containers = []corev1.Container{
-		{
-			Name:    "openstackclient",
-			Image:   instance.Spec.ContainerImage,
-			Command: []string{"/bin/sleep"},
-			Args:    []string{"infinity"},
-			SecurityContext: &corev1.SecurityContext{
-				RunAsUser:  &runAsUser,
-				RunAsGroup: &runAsGroup,
-			},
-			Env: []corev1.EnvVar{
-				{
-					Name:  "OS_CLOUD",
-					Value: "default",
-				},
-				{
-					Name:  "CONFIG_HASH",
-					Value: configHash,
-				},
-				{
-					Name:  "SECRET_HASH",
-					Value: secretHash,
-				},
-			},
-			VolumeMounts: []corev1.VolumeMount{
-				{
-					Name:      "openstack-config",
-					MountPath: "/etc/openstack/clouds.yaml",
-					SubPath:   "clouds.yaml",
-				},
-				{
-					Name:      "openstack-config-secret",
-					MountPath: "/etc/openstack/secure.yaml",
-					SubPath:   "secure.yaml",
+	clientContainer := corev1.Container{
+		Name:    "openstackclient",
+		Image:   instance.Spec.ContainerImage,
+		Command: []string{"/bin/sleep"},
+		Args:    []string{"infinity"},
+		SecurityContext: &corev1.SecurityContext{
+			RunAsUser:                ptr.To[int64](42401),
+			RunAsGroup:               ptr.To[int64](42401),
+			RunAsNonRoot:             ptr.To(true),
+			AllowPrivilegeEscalation: ptr.To(false),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{
+					"ALL",
 				},
 			},
 		},
```

Comment:
could add this to lib-common as its probably used by deployments across operators